dspace-server-webapp/src/test/java/org/dspace/app/rest/OrcidExternalSourcesIT.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static com.jayway.jsonpath.matchers.JsonPathMatchers.hasJsonPath;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.services.ConfigurationService;
+import org.hamcrest.Matchers;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * This test suite includes end to end test to verify the integration with ORCID as an External Source.
+ * The test run only if the orcid.clientid property is configured.
+ * 
+ * @author Mykhaylo Boychuk (4Science.it)
+ *
+ */
+public class OrcidExternalSourcesIT extends AbstractControllerIntegrationTest {
```

Comment:
I'd like to see a second Integration Test that runs against a sample (static) response from ORCID (which we can enable to run always in Travis CI).  Otherwise, we only have this disabled end-to-end test, which wouldn't allow us to continually validate that our code works with the ORCID API.

Reply:
It looks like this was resolved by adding some static XML test output and adding a few `*MockitoTest()` methods to this same class.

@abollini : Could you update the JavaDocs at the top of this class to note that it does *static* testing by default (using the `*MockitoTest()` methods below), but will do full end-to-end testing (of the real API) when you set `orcid.clientid`